chain/x/zoracle/internal/types/msgs.go

```diff
@@ -135,6 +135,13 @@ func (msg MsgReportData) ValidateBasic() sdk.Error {
 	if msg.DataSet == nil || len(msg.DataSet) == 0 {
 		return ErrInvalidBasicMsg("MsgReportData: Data set must not be empty.")
 	}
+	for idx1, rawReport1 := range msg.DataSet {
+		for _, rawReport2 := range msg.DataSet[idx1+1:] {
+			if rawReport1.ExternalDataID == rawReport2.ExternalDataID {
+				return ErrInvalidBasicMsg("MsgReportData: External IDs in dataset must be unique.")
+			}
+		}
+	}
```

Comment:
Can we check unique by use map? I think that way is easy to read and better in time complexity